### PR TITLE
Implementing text

### PR DESCRIPTION
**Pull request by simutaroman**

### CODE REVIEW

**Comment by simutaroman** [comment]
@intronate67 Hi
Basic implementation for extruded text is ready
image
Tomorrow I'll add group support (text can be group child, not only scene itself)
I think this will be enough for this PR. Supporting shapes and stokes will be added a little bit later

File: src/dotnet/Blazor3D/Blazor3D/Geometires/TextGeometry.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        /// <summary>
+        /// How far from text outline bevel starts. Default is 0.
+        /// </summary>
+        public double BevelOffset { get; set; } = 0;
+
+        /// <summary>
+        /// Number of bevel segments
+        /// </summary>
+        public int BevelSegments { get; set; } = 3;
+
```
Comment (simutaroman) [inline src/dotnet/Blazor3D/Blazor3D/Geometires/TextGeometry.cs:60]: remove blank lines

File: src/javascript/Viewer/Viewer3D.js [hunk 2]
```diff
@@ -129,10 +130,17 @@ class Viewer3D {
     this.scene.uuid = this.options.scene.uuid;
 
     this.options.scene.children.forEach((childOptions) => {
-      var child = SceneBuilder.BuildChild(childOptions, this.scene);
-      if (child) {
-        this.scene.add(child);
+      if (childOptions.type == "Text") {
+
```
Comment (simutaroman) [inline src/javascript/Viewer/Viewer3D.js:134]: blank line

File: src/javascript/Builders/TextBuilder.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+          });
+          const material = MaterialBuilder.buildMaterial(options.material);
+          const mesh = new THREE.Mesh(geometry, material);
+          mesh.uuid = options.uuid;
+          Transforms.setPosition(mesh, options.position);
+          Transforms.setRotation(mesh,options.rotation);
+          Transforms.setScale(mesh,options.scale);
+          parent.add(mesh);
+
```
Comment (simutaroman) [inline src/javascript/Builders/TextBuilder.js:37]: remove empty line

File: src/javascript/Builders/TextBuilder.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      },
+    
+      // onProgress callback
+      function ( xhr ) {
```
Comment (simutaroman) [inline src/javascript/Builders/TextBuilder.js:41]: null

**Comment by intronate67** [comment]
Looking good, I can already see an easy way to get ShapeGeometry working. Though, figuring out how to do the strokes will be some effort.

File: src/javascript/Builders/TextBuilder.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import MaterialBuilder from "./MaterialBuilder";
+import { FontLoader } from 'three/examples/jsm/loaders/FontLoader.js';
+import { TextGeometry } from 'three/examples/jsm/geometries/TextGeometry';
+
+class TextBuilder {
+  static BuildText(options, parent) {
+
+    const loader = new FontLoader();
+    
+    loader.load(
```
Comment (intronate67) [inline src/javascript/Builders/TextBuilder.js:12]: Do we really want to load each font for each TextGeometry? 

Would it possibly more efficient to store each the unique fonts in memory in the case where multiple TextGeometries need to be constructed? This way the method could reference a font that has already been loaded or load it just in time if we haven't already.

**Comment by simutaroman** [comment]
> Looking good, I can already see an easy way to get ShapeGeometry working. Though, figuring out how to do the strokes will be some effort.

Yes, you're right. It's very easy to implement. Please, see updates to this PR

image

**Comment by simutaroman** [comment]
Basically, it's done. Strokes are a little bit tricky in JS part, but it works

image

**Comment by simutaroman** [comment]
@intronate67  I need to finish this PR, If there are any questions or suggestions, feel free to write them down here

I suspect, there will be troubles with returning added text object ids to .NET side. But for now, it's OK

**Pull request merged**